inhibit/inhibit.go

```diff
@@ -201,26 +182,16 @@ func NewInhibitRule(cr *config.InhibitRule) *InhibitRule {
 		SourceMatchers: sourcem,
 		TargetMatchers: targetm,
 		Equal:          equal,
-		scache:         map[model.Fingerprint]*types.Alert{},
+		// Get the correct ctx in here.
+		scache: store.NewAlerts(context.Background(), 15*time.Minute),
```

Comment:
I really want to get the correct `ctx` in here, so that when `Stop()` is called on `inhibitor`, it propagates down to the store running in each inhibitrule, but I'm having trouble figuring out how best to do it. The separation currently between creating the inhibitor and its inhibit rules, and running the inhibitor, makes sense. The alerts store does creation and starts running its GC in the same creation step, which my gut says is right, but is at odds with the separation in the inhibitor. 

Any ideas how to fix this? I don't want to pass the `ctx` into the inhibitor creation function and then add that as a struct member, i think that's against "best practices", but barring that, I'm not sure what to do. It's an equally bad idea to defer inhibit rule creation until `Run()` is called? I'm not sure.

Reply:
@mxinden @simonpasquier

Reply:
I guess `store.Alerts.run` could be delayed to be triggered by `Run()` of the inhibitor. Why not call a `Stop` function on all alert stores inside `func (ih *Inhibitor) Stop() {`?

Reply:
I guess we could do that, I just like how the store is relatively simple and the user doesn't have to deal with a "stop" or a "start". I'll add it to move this forward, I want this to be done.